src/main/java/org/kizombadev/markdownparser/SyntaxAnalyzer.java

```diff
@@ -0,0 +1,161 @@
+
+        handleLineContainer(bigHeadline);
+        currentRoot.addChild(bigHeadline);
+    }
+
+    private void handleSmallHeadlineLine(Syntax currentRoot) {
+        Syntax smallHeadline = new SmallHeadline();
+        stepTokenForward();
+
+
+        handleLineContainer(smallHeadline);
+        currentRoot.addChild(smallHeadline);
+    }
+
+    private void handleLineContainer(Syntax currentRoot) {
+
+        while (currentToken() != null && !currentToken().equals(Token.NewLine)) {
+
+            if (currentToken() != null && currentToken().equals(Token.DoubleStar)) {
+                if (!stack.isEmpty() && stack.peek().equals(State.BoldBegin)) {
+                    stack.push(State.BoldEnd);
+                    return;
+                } else {
+                    handleBold(currentRoot);
+                }
+            } else if (currentToken() != null && currentToken().equals(Token.Star)) {
+                if (!stack.isEmpty() && stack.peek().equals(State.ItalicBegin)) {
+                    stack.push(State.ItalicEnd);
+                    return;
+                } else {
+                    handleItalic(currentRoot);
+                }
+            } else if (currentToken() != null && currentToken().isTextToken()) {
+                currentRoot.addChild(new TextSyntax());
+                stepTokenForward();
+            }
+        }
+    }
+
+    private void handleBold(Syntax currentRoot) {
+        stepTokenForward();
+
+
+        stack.push(State.BoldBegin);
+        TempContainer tempContainer = new TempContainer();
+        handleLineContainer(tempContainer);
+
+        if (stack.peek().equals(State.BoldEnd)) {
+            Syntax bold = new BoldSyntax();
+
+            for (Syntax child : tempContainer.getChildren()) {
+                bold.addChild(child);
+            }
+            currentRoot.addChild(bold);
+
+            stack.pop();
+            stack.pop();
+        }
+    }
+
+    private void handleItalic(Syntax currentRoot) {
+        stepTokenForward();
+
+
+        stack.push(State.ItalicBegin);
+        TempContainer tempContainer = new TempContainer();
+        handleLineContainer(tempContainer);
+
+        if (stack.peek().equals(State.ItalicEnd)) {
+            Syntax italic = new ItalicSyntax();
+
+            for (Syntax child : tempContainer.getChildren()) {
+                italic.addChild(child);
+            }
+            currentRoot.addChild(italic);
+
+            stack.pop();
+            stack.pop();
+        }
+    }
+
+    private Token currentToken() {
+        if (tokenIndex >= tokens.size()) {
+            return null;
+        }
+
+        return tokens.get(tokenIndex);
+    }
+
+    private Token nextToken() {
+        if (tokenIndex + 1 >= tokens.size()) {
+            return null;
+        }
+
+        return tokens.get(tokenIndex + 1);
+    }
+
+    private void stepTokenForward() {
+        tokenIndex++;
+    }
+
+    private enum State {
+        BoldBegin,
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00115)